### PR TITLE
Feature/results in memory

### PR DESCRIPTION
**Pull request by meganrm**
Problem
=======
In working on the variable sized ingredient setting we realized how many different places and ways the packed data was being stored. See [this wiki ](https://github.com/mesoscope/cellpack/wiki/storing-packing-results-in-memory)

Solution
========
We created a new class PackedObjects. At first I didn't store the ingredient in the object, but it ended up making more sense to do it (I was calling "get_ingredient_by_name" a lot). 

Other changes:
1. Delete unused functions 
2. Move functions into Environment if all the code was related to that class
3. change `compNum` to `compartment_id`

with @mogres 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not work as expected)

Steps to Verify:
----------------
1. `pack -r '/Users/meganr/dev/mesoscope/cellpack/cellpack/tests/recipes/v2/test_variable_size.json'`

Screenshots (optional):
-----------------------
Screenshot 2023-09-21 at 4 28 20 PM

### CODE REVIEW

File: cellpack/autopack/plotly_result.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def add_square(self, side_length, pos, rotMat, color, opacity=1):
         )
 
     def add_ingredient_positions(self, env):
-        for pos, rot, ingr, ptInd in env.molecules:
+        for pos, rot, ingr, ptInd, radius in env.molecules:
```
Comment (meganrm) [inline cellpack/autopack/plotly_result.py:62]: - [ ] test and fix 2D plotly results

**Review by mogres — COMMENTED** [review]
This was breaking image export so I made some changes to fix that. Looks good otherwise and a super useful update going forward!

File: cellpack/autopack/Graphics.py [hunk 2]
```diff
@@ -1302,7 +1302,7 @@ def displayCytoplasmIngredients(self):
         else:
             return
         if self.doSpheres:
-            for pos, rot, ingr, ptInd in self.histo.molecules:
+            for pos, rot, ingr, ptInd, _ in self.histo.molecules:
```
[the remark this answers is not in the document]
Comment (meganrm) [reply cellpack/autopack/Graphics.py:1305]: we're not, I can delete it in a separate simple PR

File: cellpack/autopack/ingredient/single_sphere.py [hunk 3]
```diff
@@ -96,6 +98,28 @@ def __init__(
         self.encapsulating_radius = radius
         self.min_radius = radius
 
+    @staticmethod
+    def create_voxelization(
+        self, image_data, bounding_box, voxel_size, image_size, position, rotation, radius, hollow=None, mesh_store=None,
```
[the remark this answers is not in the document]
Comment (meganrm) [reply cellpack/autopack/ingredient/single_sphere.py]: I reverted this, we did this because we weren't originally storing the ingredient

**Comment by mogres** [comment]
I also made a couple of minor changes to get partner packing to work with variable size since I needed it to create some synthetic data.

File: cellpack/autopack/plotly_result.py (hunk 1 above)
Comment (meganrm) [reply cellpack/autopack/plotly_result.py:62]: - [ ] 3d sphere tree looks questionable

**Review by mogres — APPROVED** [review]
New updates look good!

- Image export is working with compartments
- 2D plots are working
- Sphere tree packing is working

**Comment by mogres** [comment]
Closes #178

**Pull request merged**